### ISSUE TITLE
Enhancement wishes

### ISSUE THREAD

**Issue by khassel** [ianperrin/MMM-Strava#5]
Hi Ian,

thank you for your strava module! 

I‘m building my first MM and your module is a must for a cyclist.

There are 2 wishes.

In Table-mode I miss the altitude difference:

![grafik](https://cloud.githubusercontent.com/assets/25914086/23238035/5974b648-f960-11e6-8f89-5d80b4ac131a.png)

Second wish: My girlfriend is cyclist too. So I tried a second module in MM, but this doesn‘t work. To compare 2 strava profiles I appreciate to see e.g. the cycle activities one below the other:

![grafik](https://cloud.githubusercontent.com/assets/25914086/23238043/6454500a-f960-11e6-9bac-0a27bcdae5a4.png)

The screenshots are not build with photoshop, I changed your code already … and modified the following files:

node_helper.js
StravaAPI.js
MMM-Strava.js

So if you are interested in this changes let me know how to push them to your repository.

Thanks,

Karsten.

P.S.: Sorry for my bad english ...

**Comment by Papi77** [comment]
Hi, would it be possible to get the 3 files You mentioned above? I have new raspberry pi 3 with new OS - and have same issues: the strava module starts to flicker and/or disappears after 10-15 mins.
This issues still not fxied?

**Comment by khassel** [comment]
Hi, I forked the original-repo and you find the updated version [here](https://github.com/khassel/MMM-Strava/tree/fix_singleton_api).
Karsten.

**Comment by ianperrin** [comment]
Thanks  @khassel 

If you want to submit a pull request we can incorporate the changes here

**Comment by khassel** [comment]
done. But the Travis CI build failed ... May take a look into the code, I'm not so familiar with js.
Thanks,
Karsten.

**Comment by Papi77** [comment]
Hi, the I have updated the module this morning but the flickering/disappearing issue still persists. Also, I don't see the elevation data in "table" mode (should we make changes in config.js file?). The "chart" mode still doesn't work for me - it just renders the module into "Loading" state all the time :(

**Comment by khassel** [comment]
Hi, I think you fetched my old master branch ... i updated this a few minutes ago, so it now points to the latest commit. So please fetch again. Your config is O.K.
Karsten.

**Comment by Papi77** [comment]
Yep, I fetched it again and this time I see the elevation column as expected! I will check if the issue with the flickering / disappearing is still there over weekend... (it's midnight my timezone already ;)
Many thanks for Your files!

**Comment by khassel** [comment]
Chart mode works in my configurations. Did you use your config-code from above plus the line

mode: 'chart',

? 
Is your access_token correct?

Ian has merged my code in his repo, so you should use his original repo ...

Karsten.

**Comment by Papi77** [comment]
Hi, after two days of running both the elevation and flickering issue seems to be fixed! Cool!
Regarding the chart mode, I am using the following in my config.js:
                {
                        module: 'MMM-Strava',
                        header: 'Strava',
                        position: 'top_left',
                        config: {
                                strava_id: '<id>',
                                access_token: '<token>',
                                //elevation: true,
                                //auto_rotate: true,
                                mode: 'chart'
                                }
                },

This gives me "Loading ..." status which never completes. I believe I'm using correct access_token as I'm able to view the information in "table' mode without any issues...

**Comment by khassel** [comment]
Sorry, no ideas to chart-problem, can't reproduce it ...

**Comment by Papi77** [comment]
One thing I noticed when viewing pm2 log mm...
When the mode is set to "table", in the log file I see:

> 0|mm       | MMM-Strava received a notification: CONFIG
> 0|mm       | MMM-Strava is fetching athlete stats
> 0|mm       | 2017-04-02T20:45:46.071Z Make request: /api/v3/athletes/<id>/stats (GET) accessToken=<token>
> 

However, in "chart" mode I see:

> 0|mm       | MMM-Strava received a notification: CONFIG
> 0|mm       | MMM-Strava is fetching athlete activity after 1490565600
> 0|mm       | 2017-04-02T20:46:59.502Z Make request: /api/v3/athlete/activities?after=1490565600 (GET) accessToken=<token>

**Comment by khassel** [comment]
Thats O.K., different API calls to strava.

**Comment by Papi77** [comment]
Found the cause of the problem! If I set my language to 'en' I am able to see the chart mode. However, when I have it set to 'pl' - it doesn't work.
So, would it be possible to enable the "chart" mode for other languages?

**Comment by khassel** [comment]
'pl' works in my environment ... so still nothing to debug ...
Working with MagicMirror Version v.2.1.1

**Comment by Papi77** [comment]
If You set Your language of the whole config.js to 'pl' - what letters do You see under the days of the week?
In my case if my language is set to 'pl' globally (so that my date is in Polish):
```
var config = {
        port: 8080,
        ipWhitelist: ["192.168.12.1/24", "127.0.0.1", "::ffff:127.0.0.1", "::1"],
        language: **'pl',**

```
the chart mode doesn't work. This could make sense as in Polish the letters describing days of the week would be different... e.g. N P W Ś C P S

**Comment by khassel** [comment]
![grafik](https://cloud.githubusercontent.com/assets/25914086/24591316/116351e4-17ff-11e7-9e23-dc77ed0cf7cf.png)

**Comment by Papi77** [comment]
Hmm... that looks correct! Could this be related to my pi's environmental settings?
pi@raspberrypi3:~/MagicMirror/config $ cat /etc/default/locale
```
#  File generated by update-locale
LANG=pl_PL.UTF-8

```

**Comment by Papi77** [comment]
Got it! I ran _raspi-config_ and changed localization options (option 4). I chose to change locale (option 1) and selected the following:
```
[*] pl_PL ISO-8859-2 
[*] pl_PL.UTF-8 UTF-8 
```
-> After generating some localization files and rebooting my pi all works well!

So, I'd like to thank You **khassel** for for Your time and help on this issue. Your checks and comments allowed me finally to pin-point and fix the issue. Very much appreciated!
